components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java

```diff
@@ -169,4 +178,78 @@ private static Instant parseDateTime(String dateTimeString) {
         }
         return localDateTime;
     }
+
+    /**
+     * Get the log masking config value from config file.
+     */
+    public static void getLogMaskingConfigValue() {
+
+        isLogMaskingEnable = Boolean.parseBoolean(IdentityUtil.getProperty(ENABLE_LOG_MASKING));
+    }
+
+    /**
+     * Util function to mask content.
+     *
+     * @param content Content that needs to be masked.
+     * @return masked content.
+     */
+    public static String maskContent(String content) {
+
+        if (StringUtils.isNotEmpty(content)) {
+            Pattern pattern = Pattern.compile(LogConstants.CONTENT_MASKING_REGEX);
+            content = pattern.matcher(content).replaceAll(LogConstants.MASKING_CHARACTER);
+        }
+        return content;
+    }
+
+    /**
+     * Util function to mask claim values except userid claim.
+     *
+     * @param claims Map of user claims.
+     * @return sanitized map of user claims.
+     */
+    public static Map<String, String> getMaskedClaimsMap(Map<String, String> claims) {
+
+        Map<String, String> sanitizedClaims = new HashMap<>();
+        if (MapUtils.isNotEmpty(claims)) {
+            for (Map.Entry<String, String> entry : claims.entrySet()) {
+                if (LogConstants.USER_ID_CLAIM_URI.equals(entry.getKey())) {
+                    sanitizedClaims.put(entry.getKey(), entry.getValue());
+                } else {
+                    sanitizedClaims.put(entry.getKey(), maskContent(entry.getValue()));
+                }
+            }
+        }
+        return sanitizedClaims;
+    }
+
+    /**
+     * Util function to mask claim value except userId claim when there is only one claim.
+     *
+     * @param claimURI   Claim URI.
+     * @param claimValue Claim value that will be masked.
+     * @return masked claim value.
+     */
+    public static String maskClaimValue(String claimURI, String claimValue) {
+
+        if (StringUtils.isNotBlank(claimValue) && !LogConstants.USER_ID_CLAIM_URI.equals(claimURI)) {
```

Comment:
`StringUtils.isNotBlank(claimValue)` -> do we really need this check here? for the map logic above such validation hasn't been done. better to have a consistent approach